README.md

```diff
@@ -1,86 +1,125 @@
+build MIGraphX source code, then describe each of the two approaches.
+
+### List of prerequisites
+The following is a list of prerequisites required to build MIGraphX source. 
 
-## Prerequisites
 * [ROCm cmake modules](https://github.com/RadeonOpenCompute/rocm-cmake) **required**
 * [MIOpen](https://github.com/ROCmSoftwarePlatform/MIOpen) for running on the GPU
+* [rocBLAS](https://github.com/ROCmSoftwarePlatform/rocBLAS) for running on the GPU
 * [HIP](https://github.com/ROCm-Developer-Tools/HIP) for running on the GPU
 * [Protobuf](https://github.com/google/protobuf) for reading [onnx](https://github.com/onnx/onnx) files
 * [Half](http://half.sourceforge.net/) - IEEE 754-based half-precision floating point library
 * [pybind11](https://pybind11.readthedocs.io/en/stable/) - for python bindings
+* [JSON](https://github.com/nlohmann/json) - for model serialization to json string format
+* [MessagePack](https://msgpack.org/index.html) - for model serialization to binary format
 
-## Installing the dependencies
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-Dependencies can be installed using the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
-
-To install rbuild:
+In this approach, we need to install the [rbuild](https://github.com/RadeonOpenCompute/rbuild) first, then use it to
+build MIGraphX. rbuild can be installed as (sudo may be needed):
 ```
-pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`.
+We also need to install [rocm-cmake](https://github.com/RadeonOpenCompute/rocm-cmake), rocblas, and miopen-hip as 
+`sudo apt install -y rocm-cmake rocblas miopen-hip`.
+
+Then MIGraphX can be built as:
 
-To build dependencies along with MIGraphX
 ```
-rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++
+rbuild build -d depend -B build --cxx=/opt/rocm/llvm/bin/clang++
 ```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
 
-## Building MIGraphX from source
+Note that for ROCm3.7 and later releases, Ubuntu 18.04 or later releases are needed. 
+Upgrapde to Ubuntu 18.04 is available at [Upgrade Ubuntu to 18.04](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/wiki/Upgrade-to-Ubuntu-18.04-for-ROCM3.7-or-later-releases)
 
-## Configuring with cmake
+#### Use cmake to build MIGraphX
 
-First create a build directory:
+In this approach, we need to install the prerequisites, configure the cmake, and then build the source.
 
+##### Installing the prerequisites
 
-```
-mkdir build; 
-cd build;
-```
+For convenience, the prerequisites can be built automatically with rbuild as:
+```rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++```, all the prerequisites are in the 
+folder `depend`. 
```

Comment:
It also should be mentioned that migraphx will be built in the `build` directory.